api_spec.yaml

```diff
@@ -107,8 +107,98 @@ paths:
         
         '500':
           $ref: '#/components/responses/500'
+          
+    get:
+      tags:
+        - OAuth2 Recipe
+      operationId: getOAuth2AuthorizationCode
+      description: |
+        get authorization code details belonging to a client.
```

Comment:
make a list of all the status types that are returned across all the APIs and post it as a reply to this comment

Reply:
list of error status types:

1.  POST `/appid-<appId>/<tenantId>/recipe/oauth2/authorizationcode`
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found
- UNKNOWN_SESSION_HANDLE_ERROR - if session_handle is not found or expired
- UNKNOWN_REDIRECT_URI_ERROR - if redirect uri is not found list of client's redirect uri's.
- UNKNOWN_SCOPE_ERROR - if scope is not found in client's allowed scope list.

3.  GET  `/appid-<appId>/<tenantId>/recipe/oauth2/authorizationcode`
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found
- UNKNOWN_AUTH_CODE_ERROR - if auth_code is not found or expired

4. DELETE `/appid-<appId>/<tenantId>/recipe/oauth2/authorizationcode` 
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found
- UNKNOWN_SESSION_HANDLE_ERROR - if session_handle is not found or expired

5. POST `/appid-<appId>/<tenantId>/recipe/oauth2/token`
- UNKNOWN_AUTH_CODE_ERROR - if auth_code is not found or expired
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found
- UNKNOWN_CLIENT_SECRET_ERROR - if client secret is not found 
- UNKNOWN_REDIRECT_URI_ERROR - if redirect uri is different from the one used for auth code creation
- UNKNOWN_REFRESH_TOKEN_ERROR - if refresh token is not found
- UNKNOWN_SCOPE_ERROR - if scope is not found in client's allowed scope list.
- SCOPE_NOT_ALLOWED_ERROR - if scope/s is not subset of scopes in auth_code table
- UNKNOWN_CODE_VERIFIER_ERROR - if code verifier doesn't match code challenge.

6. DELETE `/appid-<appId>/<tenantId>/recipe/oauth2/token`
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found
- UNKNOWN_SESSION_HANDLE_ERROR - if session_handle is not found or expired
- INVALID_ACCESS_TOKEN_ERROR - if access token is not found

7. Get `/appid-<appId>/<tenantId>/recipe/oauth2/token`
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found
- UNKNOWN_SESSION_HANDLE_ERROR - if session_handle is not found or expired
- UNKNOWN_ACCESS_TOKEN_ERROR - if access token is not found
- UNKNOWN_REFRESH_TOKEN_ERROR - if refresh token is not found

8. POST `/appid-<appId>/recipe/oauth2/scope`
- SCOPE_ALREADY_EXISTS_ERROR - if scope already exists for an app

9. PUT `/appid-<appId>/recipe/oauth2/scope`
- SCOPE_NOT_FOUND_ERROR - if the scope to be replaced is not found
- SCOPE_ALREADY_EXISTS_ERROR - if scope already exists for an app

10. POST `/appid-<appId>/recipe/oauth2/client/scope`
- UNKNOWN_SCOPE_ERROR -  if scope is not found 
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found
- SCOPE_ALREADY_EXISTS_ERROR - if client's allowed scope list already contains the scope.

11. Delete `SCOPE_ALREADY_EXISTS_ERROR `
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found

12. Post `/appid-<appId>/recipe/oauth2/client`
- UNKNOWN_SCOPE_ERROR -  if scope is not found 
- CLIENT_ALREADY_EXISTS_ERROR - if a client exists with same name.

13. Put `/appid-<appId>/recipe/oauth2/client`
- UNKNOWN_SCOPE_ERROR -  if scope is not found 
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found
- CLIENT_ALREADY_EXISTS_ERROR - if a client exists with same name.

14. Delete `/appid-<appId>/recipe/oauth2/client`
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found

15. get `/appid-<appId>/recipe/oauth2/client`
- UNKNOWN_CLIENT_ID_ERROR - if clientid is not  found

Reply:
list of error code:

`UNKNOWN_CLIENT_ID_ERROR
UNKNOWN_SESSION_HANDLE_ERROR
UNKNOWN_REDIRECT_URI_ERROR
UNKNOWN_SCOPE_ERROR
UNKNOWN_AUTH_CODE_ERROR
UNKNOWN_CLIENT_SECRET_ERROR
UNKNOWN_REFRESH_TOKEN_ERROR
SCOPE_NOT_ALLOWED_ERROR
UNKNOWN_CODE_VERIFIER_ERROR
INVALID_ACCESS_TOKEN_ERROR
UNKNOWN_ACCESS_TOKEN_ERROR
SCOPE_ALREADY_EXISTS_ERROR
SCOPE_NOT_FOUND_ERROR
CLIENT_ALREADY_EXISTS_ERROR`

Reply:
- Whats the diff between UNKNOWN_SCOPE_ERROR and SCOPE_NOT_ALLOWED_ERROR?
- INVALID_ACCESS_TOKEN_ERROR vs UNKNOWN_ACCESS_TOKEN_ERROR
- SCOPE_NOT_FOUND_ERROR vs UNKNOWN_SCOPE_ERROR

Reply:
Changes made.